designer/server/src/main/scala/pl/touk/nussknacker/ui/process/test/testdataformat/CommonDataFormatHandler.scala

```diff
@@ -0,0 +1,74 @@
+package pl.touk.nussknacker.ui.process.test.testdataformat
+
+import cats.implicits.toBifunctorOps
+import io.circe.parser
+import io.circe.syntax.EncoderOps
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.json.encoders.{StrictToJsonEncoder, ToJsonEncoder}
+import pl.touk.nussknacker.engine.api.livedata.LiveDataProvider
+import pl.touk.nussknacker.engine.api.process.Source
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.ui.process.test.{
+  CommonFormatPreliminaryScenarioRecord,
+  PreliminaryScenarioRecord,
+  PreliminaryScenarioRecords,
+  SerializedScenarioRecordsContent
+}
+import pl.touk.nussknacker.ui.process.test.testdataformat.TestDataFormatSerDe.DeserializationError
+import shapeless.syntax.typeable.typeableOps
+
+class CommonDataFormatHandler(modelData: ModelData) extends TestDataFormatHandler {
+
+  override val serDe: TestDataFormatSerDe = CommonDataFormatSerDe
+
+  private val toJsonEncoder = new StrictToJsonEncoder(modelData.modelClassLoader)
+
+  override def canFetchLiveData(compiledSource: Source): Boolean = compiledSource.isInstanceOf[LiveDataProvider]
+
+  override def canBeTested(compiledSource: Source): Boolean =
+    true // For common format, every source can be used for scenario testing
+
+  override def fetchLiveData(
+      sourceId: NodeId,
+      compiledSource: Source,
+      maxNumberOfRecords: Int
+  ): Either[TestDataFormatHandler.LiveDataFetchingNotSupportedError.type, List[PreliminaryScenarioRecord]] = {
+    compiledSource
+      .cast[LiveDataProvider]
+      .map { liveDataProvider =>
+        val records = modelData.withModelClassloaderAsContextClassLoader {
+          val sourceRecords = liveDataProvider.fetchLiveData(maxNumberOfRecords).records
+
+          sourceRecords
+            .map { record =>
+              val variablesAsJson = record.variables.mapValuesNow(toJsonEncoder.encodeUnsafe)
+              CommonFormatPreliminaryScenarioRecord(sourceId.id, variablesAsJson, record.timestamp)
+            }
+        }
+        Right(records)
+      }
+      .getOrElse(Left(TestDataFormatHandler.LiveDataFetchingNotSupportedError))
```

Comment:
how about simple pattern match like:

```scala
compiledSource match {
  case liveDataProvider: LiveDataProvider => ...
  case _ =>
      Left(TestDataFormatHandler.LiveDataFetchingNotSupportedError)
```

same for SourceSpecificDataFormatHandler